src/toil/wdl/wdl_analysis.py

```diff
@@ -620,18 +655,35 @@ def parse_declaration_type(self, typeAST):
         which Toil needs to import, so we recursively travel down to the innermost
         type which will tell us if the variables are files that need importing.
 
+        For Pair and Map compound types, we recursively travel down the subtypes and
+        store them as attributes of a `WDLType` string. This way, the type structure is
+        preserved, which will allow us to import files appropriately.
+
```

Comment:
The `Currently supported` section above should be updated with `Pair` and also I think `Map`?